server/portal/apps/auth/backends.py

```diff
@@ -27,21 +28,22 @@ def authenticate(self, *args, **kwargs):
                 tapis_user = json_result['result']
                 username = tapis_user['username']
                 UserModel = get_user_model()
+
                 try:
-                    user = UserModel.objects.get(username=username)
-                    user.first_name = tapis_user['given_name']
-                    user.last_name = tapis_user['last_name']
-                    user.email = tapis_user['email']
-                    user.save()
-                except UserModel.DoesNotExist:
-                    logger.info('Creating local user record for "%s" '
-                                'from Tapis Profile' % username)
-                    user = UserModel.objects.create_user(
-                        username=username,
-                        first_name=tapis_user['given_name'],
-                        last_name=tapis_user['last_name'],
-                        email=tapis_user['email']
-                    )
+                    user_data = get_user_data(username=username)
+                    defaults = {
+                        'first_name': user_data['firstName'],
+                        'last_name': user_data['lastName'],
+                        'email': user_data['email']
+                    }
+                except Exception as e:
+                    logger.exception(e)
+                    return user
+
+                user, created = UserModel.objects.update_or_create(username=username, defaults=defaults)
```

Comment:
Out of curiosity, is `update_or_create()` used here in the case that the user info has changed in TAS and we want to keep it up to date on our end?

Reply:
Yes, exactly - we'll update the profile every time they login